iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/function_ref.inl

```diff
@@ -81,10 +107,13 @@ inline void function_ref<ReturnType(ArgTypes...)>::swap(function_ref<ReturnType(
 }
 
 template <class ReturnType, class... ArgTypes>
+// AXIVION Next Line AutosarC++19_03-A2.10.4 : Overload for swap(function_ref, function_ref) as in STL
 inline void swap(function_ref<ReturnType(ArgTypes...)>& lhs, function_ref<ReturnType(ArgTypes...)>& rhs) noexcept
```

Comment:
Could you please declare this in the header. This is public API but declared and defined hidden in the `inl` file. But the include of the inl file at the end of the header makes this again public.

And as you stated this is also defined in the STL so it absolutely makes sense to provide the same functionality correctly to the user.

Reply:
Sure, fixed!